modules/tinymce/src/plugins/table/test/ts/browser/ui/TableDialogStyleWithCssTest.ts

```diff
@@ -0,0 +1,200 @@
+import { ApproxStructure } from '@ephox/agar';
+import { context, describe, it } from '@ephox/bedrock-client';
+import { Arr, Obj } from '@ephox/katamari';
+import { TinyAssertions, TinyHooks, TinySelections } from '@ephox/mcagar';
+import { Attribute, Css, Html, SelectorFilter, SugarElement } from '@ephox/sugar';
+
+import Editor from 'tinymce/core/api/Editor';
+import Plugin from 'tinymce/plugins/table/Plugin';
+import Theme from 'tinymce/themes/silver/Theme';
+
+import * as TableTestUtils from '../../module/test/TableTestUtils';
+
+describe('browser.tinymce.plugins.table.ui.TableCellDialogTest', () => {
+  const generalSelectors = {
+    cellspacing: 'label.tox-label:contains(Cell spacing) + input.tox-textfield',
+    cellpadding: 'label.tox-label:contains(Cell padding) + input.tox-textfield'
+  };
+
+  const initializeTable = (editor: Editor, tableAttributes: Record<string, string>, tableStyles: Record<string, string>, cellStyles: Record<string, string>) => {
+    const defaultTable = SugarElement.fromHtml<HTMLTableElement>(
+      '<table>' +
+      '<tbody>' +
+      '<tr>' +
+      '<td><p>Hello</p></td>' +
+      '<td><p>Next cell</p></td>' +
+      '</tr>' +
+      '<tr>' +
+      '<td><p>Cell 3</p></td>' +
+      '<td><p>Last one</p></td>' +
+      '</tr>' +
+      '</tbody>' +
+      '</table>'
+    );
+
+    Attribute.setAll(defaultTable, tableAttributes);
+    Css.setAll(defaultTable, tableStyles);
+
+    Arr.each(SelectorFilter.descendants(defaultTable, 'td,th'), (elem) => Css.setAll(elem, cellStyles));
+
+    editor.setContent(Html.getOuter(defaultTable));
+    TinySelections.setCursor(editor, [ 0, 0, 0 ], 0);
+  };
+
+  const assertTable = (editor: Editor, tableAttributes: Record<string, string>, tableStyles: Record<string, string>, cellStyles: Record<string, string>) => {
```

Comment:
I'd have:
`const assertTable = (editor: Editor, spec: { tableAttributes: Record<string, string>, tableStyles: Record<string, string>, cellStyles: Record<string, string> }) => {`

So in your tests would be easier to read:

`assertTable(editor, { tableAttributes: { cellspacing: '5px' }, tableStyles: { 'border-spacing': '5px' }, cellStyles: { padding: '30px' }});`

Same for `initializeTable`.

Reply:
I went slightly further (compressed it all into one object with 4 optional fields) but this was a good catch, thanks